### PR TITLE
Implement local save and load buttons

### PR DESCRIPTION
**Pull request by mslw**
This PR implements https://github.com/psychoinformatics-de/sfb1451_org/issues/33

Added two buttons:
- _Daten Lokal Speichern_ saves the entered data as a json file (triggers download of the file). The content is based on what previously went into ContentString (used for checksumming) with minor changes. Subject pseudonym is used as the file name.
- _Lokale Daten Laden_ loads such a file and puts the content into respective fields.

I am not sure if the save / load implementation is optimal, but that's what I came up with.

I was able to save - load - save entered data, but manual testing is always welcome.

### CODE REVIEW

**Comment by christian-monch** [comment]
If "Probanden-Pseudonym" is not set, a file name `json` is locally saved.

Either saving should be refused, if "Probanden-Pseudonym" is unset, or there should be the possibility to choose a name manually in case it is unset

File: entry.html [hunk 1]
```diff
@@ -2777,16 +2789,26 @@ <h5 class="mb-3"><label for="signature-data">Signaturdaten</label></h5>
...
+	return contentString;
+    }
 
-        let contentString = valueArray.join(";");
-        console.log("contentString: " + contentString);
-        return contentString;
+    function getContentObj() {
+	let keyValueArray = getContentArray();
+	let contentObj = Object.fromEntries(keyValueArray);
+	contentObj['subject-group'] = getStringContent('subject-group');  // not present in getContentString
```
Comment (christian-monch) [inline entry.html]: Just for my understanding: why is `subject-group` not present in `getContentString()`-results?

File: entry.html [hunk 2]
```diff
@@ -3073,6 +3095,206 @@ <h5 class="mb-3"><label for="signature-data">Signaturdaten</label></h5>
 
 </script>
 
+<script>
+    // Local save
+    const lsb = document.getElementById('local-save-button');
+
+    lsb.onclick = function() {
+	let savedObject = getContentObj();
+	let fname = savedObject['subject-pseudonym'] + ".json"  // use Pseudonym as filename
```
Comment (christian-monch) [inline entry.html]: If `subject-pseudonym` is not set, `fname` will be `.json` (there is also a comment in the PR discussion). Maybe decline to save without `subject-pseudonym` or ask for a value?

File: entry.html [hunk 3]
```diff
@@ -3073,6 +3095,206 @@ <h5 class="mb-3"><label for="signature-data">Signaturdaten</label></h5>
...
+	var bb = new Blob([JSON.stringify(savedObject)], { type: "text/json;charset=utf-8" });
+	var a = document.createElement('a');
+	a.setAttribute('download', fname)
+	a.setAttribute('href', URL.createObjectURL(bb))
+	a.click();
+	return false;
```
Comment (christian-monch) [inline entry.html]: Maybe remove element `a` again?

File: entry.html [hunk 4]
```diff
@@ -3073,6 +3095,206 @@ <h5 class="mb-3"><label for="signature-data">Signaturdaten</label></h5>
...
+    const fileElem = document.getElementById("local-load-input");
+
+    fileSelect.addEventListener("click", function (e) {
+	if (fileElem) {
+	    fileElem.click();
+	}
+    }, false);
+
+    fileElem.addEventListener("change", readSavedJSON, false);
+
```
Comment (christian-monch) [inline entry.html]: Generally it would be nice if you could unify tab and space, more precisly, replace tabs with 4 spaces

**Review by christian-monch — CHANGES_REQUESTED** [review]
Generally looks good to me. Can be merged if the comments are addressed.

File: entry.html (hunk 1 above)
Comment (mslw) [reply entry.html]: It wasn't when I started. The old `getContentString` explicitly defines an array with a list of ids, and `subject-group` wasn't there. May be a simple omission.

I chose to tack it on to contentObj (for saving) rather then change the existng getContentString behaviour.

File: entry.html (hunk 2 above)
Comment (mslw) [reply entry.html]: I'm going for "decline to save" - it makes little sense to save records not attributed to any subject.

I also don't know an easy way to ask for value. I found [showSaveFilePicker](https://developer.mozilla.org/en-US/docs/Web/API/Window/showSaveFilePicker) in JS, but it's new and currently supported only by Chromium-based browsers.

File: entry.html (hunk 4 above)
Comment (mslw) [reply entry.html]: I thought I knew my editor, but apparently not... I'll clean it up (probably I'll run prettier on the script blocks).

**Comment by mslw** [comment]
I made the suggested changes (commented above). I used prettier to clean up not only tabs, but also quotes, semicolons, spaces, and the like.

**Review by christian-monch — APPROVED** [review]
LGTM, will merge
Thx a lot

**Pull request merged**